griptape/tasks/branch_task.py

```diff
@@ -0,0 +1,30 @@
+from __future__ import annotations
+
+from typing import Callable, Union
+
+from attrs import define, field
+
+from griptape.artifacts import InfoArtifact, ListArtifact
+from griptape.tasks import BaseTask, CodeExecutionTask
+
+
+@define
+class BranchTask(CodeExecutionTask):
+    on_run: Callable[[BranchTask], Union[InfoArtifact, ListArtifact[InfoArtifact]]] = field(kw_only=True)
+
```

Comment:
should we set a final "chosen" task_id on the class?

Reply:
That is the output of the task, are you envisioning something else?

Reply:
yeah i see now. maybe as a property we can grab the str value of the output? it just seems like you should be able to directly reference it, considering thats the whole point of the task.

Reply:
Agreed, but that seems like an improvement for all Artifacts, not just ones returned from this Task.

Reply:
no just the task has that property I mean

Reply:
Yeah and I'm proposing that the improvement be made to all Artifacts/Tasks. Feels a bit weird to introduce some syntactic sugar for only one type of Task.